src/api-service/__app__/onefuzzlib/proxy.py

```diff
@@ -154,6 +154,34 @@ def stopped(self) -> None:
         logging.info(PROXY_LOG_PREFIX + "removing proxy: %s", self.region)
         self.delete()
 
+    def is_outdated(self) -> bool:
```

Comment:
Once every 7 days, all sessions will terminate without warning.  The user experience is going to be pretty painful.

Instead of a hard shutdown, I propose starting a new proxy, waiting until this one is no longer used, then deleting this one.

Reply:
Is there a way to mark a proxy as 'deprecated'? So that it's not picked up by future calls to 'get_or_create'? If I don't immediately put an outdated proxy in a stopping state, and create a new proxy - how do I avoid that old proxy from being used? 

I could create an 'outdated' flag that is set when a proxy gets flagged as out of date by the is_outdated function. Then get_or_create would only retrieve proxies that aren't marked as 'outdated'.

Reply:
Okay I submitted a number of changes that might have addressed the concern here.